File: kashyyyk/src/main/java/org/example/product/kashyyyk/KashyyykModule.java
package org.example.product.kashyyyk;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import org.example.product.bespin.BespinModule;
import org.example.product.kamino.KaminoModule;
import org.example.product.naboo.NabooModule;
import org.example.product.tatooine.TatooineModule;
import android.widget.Button;

public class KashyyykModule {
    public Button button;

    public KashyyykModule() {
        try (InputStream hello = KashyyykModule.class.getResourceAsStream("hello.txt")) {
            String message = new BufferedReader(new InputStreamReader(hello)).readLine();
            System.out.println(message);
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
    }

    public NabooModule flyTo(TatooineModule from) {
        new KaminoModule();
        new BespinModule();
        int a = calculateSomethingImportant();
        return new NabooModule(a);
    }

    int calculateSomethingImportant() {
        return 4 + 1;
    }
}

File: kashyyyk/src/main/kotlin/org/example/product/kashyyyk/KashyyykModule.kt
package org.example.product.kashyyyk

import java.io.BufferedReader
import java.io.IOException
import java.io.InputStreamReader
import org.example.product.bespin.BespinModule
import org.example.product.kamino.KaminoModule
import org.example.product.naboo.NabooModule
import org.example.product.tatooine.TatooineModule

class KashyyykModule {
    init {
        try {
            KashyyykModule::class.java.getResourceAsStream("hello.txt").use { hello ->
                val message = BufferedReader(InputStreamReader(hello)).readLine()
                println(message)
            }
        } catch (e: IOException) {
            throw RuntimeException(e)
        }
    }

    fun flyTo(from: TatooineModule?): NabooModule {
        KaminoModule()
        BespinModule()
        val a = calculateSomethingImportant()
        return NabooModule(a)
    }

    fun calculateSomethingImportant(): Int {
        return 4 + 1
    }
}
